### PR TITLE
Fix #1821: Fix downloading auth token in Firefox browser

### LINKED ISSUE #1821: Download JSON  is not working in API Authentication Token pop-up box
* the observed bevaviour 

in user profile  **Download JSON** is not working
after clicking it doesn't give any response

screenshot from 2018-09-24 00-44-03

### PR DESCRIPTION
**Pull request by gautamjajoo**
Fixes #1821 .
Ref : https://bugs.jquery.com/ticket/14761
@deshraj @RishabhJain2018 

screenshot from 2019-01-11 11-43-23

### CODE REVIEW

**Comment by gautamjajoo** [comment]
@desv @RishabhJain2018 Could you review this?

File: frontend/src/js/controllers/profileCtrl.js [hunk 1]
```diff
@@ -127,7 +127,13 @@
             anchor.attr({
                 href: 'data:text/json;charset=utf-8,' + encodeURIComponent(JSON.stringify(vm.jsonResponse)),
                 download: 'token.json'
-            })[0].click();
+            })
```
Comment (RishabhJain2018) [inline frontend/src/js/controllers/profileCtrl.js]: Please add semi-colon here.

File: frontend/src/js/controllers/profileCtrl.js [hunk 2]
```diff
@@ -127,7 +127,13 @@
...
+
+            // Create Event
+            var ev = document.createEvent("MouseEvents");
+                ev.initMouseEvent("click", true, false, self, 0, 0, 0, 0, 0, false, false, false, false, 0, null);
+                // Fire event
+                anchor[0].dispatchEvent(ev);
```
Comment (RishabhJain2018) [inline frontend/src/js/controllers/profileCtrl.js]: @gautamjajoo It doesn't work on my local. Can you please verify if it is working for you.

Comment (gautamjajoo) [reply frontend/src/js/controllers/profileCtrl.js]: @RishabhJain2018 It is working on my end. I have checked it once more. Should I make a screencast?

File: frontend/src/js/controllers/profileCtrl.js (hunk 1 above)
Comment (gautamjajoo) [reply frontend/src/js/controllers/profileCtrl.js]: Done.

**Comment by gautamjajoo** [comment]
@RishabhJain2018 It is working on my end.

**Comment by RishabhJain2018** [comment]
@gautamjajoo No, it is still not working on my end. Can you please tell me which version of firefox are you using.

Also, @deshraj can you please pull this PR and test it.

**Comment by gautamjajoo** [comment]
@RishabhJain2018 I am using Firefox 64.0

**Comment by gautamjajoo** [comment]
@RishabhJain2018 Please review this also.

**Comment by RishabhJain2018** [comment]
@galipremsagar Can you please review this PR?

**Comment by gautamjajoo** [comment]
@galipremsagar Could you review this?

File: frontend/src/js/controllers/profileCtrl.js (hunk 2 above)
Comment (galipremsagar) [reply frontend/src/js/controllers/profileCtrl.js]: @RishabhJain2018 & @gautamjajoo  The changes LGTM. I've tested this out on all browsers(safari, firefox, opera, chrome on MAC OS), worked well on all of them. We can merge this.

**Comment by galipremsagar** [comment]
@RishabhJain2018 & @gautamjajoo The changes LGTM. I've tested this out on all browsers(safari, firefox, opera, chrome on MAC OS), worked well on all of them. We can merge this.

**Comment by RishabhJain2018** [comment]
It still doesn't work on my local machine :( 
@Sanji515 Can you please check this on your local machine?

**Comment by Sanji515** [comment]
@RishabhJain2018 I've tested this on `chrome & firefox` browsers on `ubuntu`, it's working on both of them. Take a look at the screenshot of firefox browser

Screenshot from 2019-05-15 18-22-18

**Comment by RishabhJain2018** [comment]
It still doesn't work on my local machine but since it works for you guys, hence I'm merging this :)

**Pull request merged**